ndn_python_repo/clients/getfile.py

```diff
@@ -38,30 +38,35 @@ async def fetch_file(self, name_at_repo: NonStrictName, local_filename: str = No
         :param local_filename: str. The filename of the retrieved file on the local file system.
         :param overwrite: If true, existing files are replaced.
         """
+
+        # If no local filename is provided, store file with last name component
+        # of repo filename
+        if local_filename is None:
+            local_filename = Name.to_str(name_at_repo)
+            local_filename = local_filename.strip().split("/")[-1]
+
+        # If the file already exists locally and overwrite=False, retrieving the file makes no
+        # sense.
+        if os.path.isfile(local_filename) and not overwrite:
```

Comment:
I wonder what should be done if the file exists but is not a regular file (e.g. directory, socket, etc.). I guess for safety we should just fail in that case, don't want to risk deleting an entire directory if `overwrite=True`...

Reply:
Looking at the Python documentation, this should not be an issue:

> os.remove() method in Python is used to remove or delete a file path. This method can not remove or delete a directory. If the specified path is a directory then OSError will be raised by the method. os.rmdir() can be used to remove directory.

Reply:
ok, sounds good!